File: SiteVisitsNetCore2020Web/ClientApp/src/app/models/City.ts
import { Region } from './Region';

export class City {
    //id: number;
  id: string;
  name: string;
  region: Region;
  constructor(options: {
    //id: number,
    id: string,
    name: string,
    region: Region
  }) {
    if (options) {
      this.id = options.id;
      this.name = options.name;
      this.region = new Region(options.region);
    }
  }
}

File: SiteVisitsNetCore2020Web/Models/City.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace SiteVisitsNetCore2020Web.Models
{
    public class City
    {
        //public int Id { get; set; }
        public Guid Id { get; set; }
        public string Name { get; set; }
        public virtual Region Region { get; set; }
        public DateTime FirstOccurrence { get; set; }
        public DateTime DateCreated { get; set; }

    }
}
